src/strategies/gadget.rs

```diff
@@ -99,8 +98,13 @@ impl<'a> Strategy<Variable> for GadgetStrategy<'a> {
         #[cfg(feature = "trace")]
         let circuit_size = self.cs.circuit_size();
 
-        let mut product = [self.cs.zero_var; WIDTH];
-        let mut z3 = self.cs.zero_var;
+        // Declare and constraint zero.
+        let zero = self.cs.add_input(BlsScalar::zero());
+        self.cs
+            .constrain_to_constant(zero, BlsScalar::zero(), BlsScalar::zero());
```

Comment:
I thought that `zero_var` was doing that when we were discussing about implementing `one_var`, or maybe I don't fully remember: is something changed and we don't have anymore the 1 constraints about zero when we create the composer?

Could you in case point me to the change in PLONK with the description to fully understand? From the changes below seems to me you're refactoring exactly how `zero_var` was used before.

Reply:
Resolved in our previous call.